src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -468,19 +468,18 @@ if this_event.type == pbs.PERIODIC:
             # nodes which are still booting, write them to
             # node file for status check in next iteration.
             if sleep_node_list or poweron_vnlist:
+                sleeping_nodes = poweron_vnlist
+                # Look for still booting nodes from previous iteration.
+                for n in sleep_node_list:
+                    if n not in ready_nodes:
+                        sleeping_nodes.append(n)
+                data = ','.join(sleeping_nodes)
                 try:
-                    fd = open(node_file, 'w')
-                    sleeping_nodes = poweron_vnlist
-                    # Look for still booting nodes from previous iteration.
-                    for n in sleep_node_list:
-                        if n not in ready_nodes:
-                            sleeping_nodes.append(n)
-                    data = ','.join(sleeping_nodes)
-                    fd.write(data)
-                    fd.close()
+                    with open(node_file, 'w') as fd:
+                        fd.write(data)
                 except IOError:
                     raise Exception("I/O error reading node file")
-                except:
+                except Exception:
```

Comment:
Same as above, just let the IOError exception be raised uncaught.

Reply:
Done.